### PR TITLE
Support for Android arm64 compilations

### PR DESCRIPTION
**Pull request by MortimerGoro**
This PR is the final step to adds support for Android arm64 compilations. See https://github.com/servo/servo/issues/11921 for previous work.

Fixes in this PR:
- Fix js dependency compilation: https://github.com/servo/rust-mozjs/pull/360
- Fix skia dependency link error: https://github.com/servo/skia/pull/136
- Fix blurdroid dependency compilation: https://github.com/szeged/blurdroid/pull/4
- Fix mio and net2 dependency compilations: https://github.com/carllerche/mio/pull/599
- Fix gcc compiler name in the fake linker
- Compile OpenSSL for aarch64 (update to stable 1.1.0 was required for this)

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [] These changes fix #__ (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17099)

### CODE REVIEW

**Comment by MortimerGoro** [comment]
cc @larsbergstrom

File: python/servo/build_commands.py [hunk 1]
```diff
@@ -257,14 +261,15 @@ def build(self, target=None, release=False, dev=False, jobs=None,
...
+            env["RUST_TARGET"] = target
             with cd(openssl_dir):
                 status = call(
                     make_cmd + ["-f", "openssl.makefile"],
                     env=env,
                     verbose=verbose)
                 if status:
                     return status
-            openssl_dir = path.join(openssl_dir, "openssl-1.0.1t")
+            openssl_dir = path.join(openssl_dir, "openssl-1.1.0f")
```
Comment (nox) [inline python/servo/build_commands.py]: We can't suddenly update openssl in the middle of nowhere of a patch that is intended to support arm64, that change is out of scope.

Comment (MortimerGoro) [inline python/servo/build_commands.py]: Ok, it makes sense to move the OpenSSL update into a separate PR. The reason for the update is that OpenSSL 1.1.0 comes with a compilation target ready for arm64, while 1.0.1 required a lot of hacking (and it's a out of support version)

**Comment by MortimerGoro** [comment]
We have decided to do the OpenSSL update into a separate PR before merging this. Let's wait for https://github.com/servo/servo/issues/17079 with the mirror and the new version.

**Comment by larsbergstrom** [comment]
Looks like #17079 landed, so this should be good to go after rebase!

**Comment by MortimerGoro** [comment]
@larsbergstrom Rebase ready! https://github.com/servo/servo/issues/17079 used OpenSSL 1.0.2 instead of the 1.1. But I found a way to compile for aarch64 in 1.0.x.

I want to add a ci build step for android64. Should I add the `./mach build` and `./mach package` commands to the `android` section in https://github.com/servo/servo/blob/master/etc/ci/buildbot_steps.yml or should I create a whole new section for android64?

**Comment by larsbergstrom** [comment]
@bors-servo r+

@MortimerGoro Please just add them to the existing builders. Adding new build sections requires a significant amount of hacking & deployment on the saltfs side and there is some time to spare in our cross-builder configurations right now.
